### PR TITLE
#28 Improve Cluster join API. Make join async.

### PR DESCRIPTION
**Pull request by myroslavlisniak**
Motivation:
* There should not be blocking calls in Cluster API

Done:
* Make transport start async
* Make cluster start async

### CODE REVIEW

**Comment by ronenhamias** [comment]
+1

File: transport/src/test/java/io/servicefabric/transport/TransportTest.java [hunk 1]
```diff
@@ -624,9 +624,13 @@ private Transport createTransport(TransportEndpoint endpoint, int sendHwm) {
...
             .useNetworkEmulator(true).build());
-    transport.start();
+      try {
+          transport.start().get();
+      } catch (Exception e) {
+          LOGGER.error("Failed to start transport ", e);
+      }
 
-    return transport;
+      return transport;
```
Comment (antonkharenko) [inline transport/src/test/java/io/servicefabric/transport/TransportTest.java]: Why did it shifted? Don't we have correct formatiing (2 spaces) before?

File: pom.xml [hunk 2]
```diff
@@ -94,6 +94,11 @@
                 <version>${rxjava.version}</version>
             </dependency>
             <dependency>
+                <groupId>io.reactivex</groupId>
+                <artifactId>rxjava-guava</artifactId>
+                <version>1.0.3</version>
```
Comment (antonkharenko) [inline pom.xml]: move version to properties

**Comment by antonkharenko** [comment]
+1

**Pull request merged**